core/services/gateway/handlers/functions/handler.functions.go

```diff
@@ -178,7 +178,16 @@ func (h *functionsHandler) HandleUserMessage(ctx context.Context, msg *api.Messa
 		return ErrRateLimited
 	}
 	if h.subscriptions != nil && h.minimumBalance != nil {
-		if balance, err := h.subscriptions.GetMaxUserBalance(sender); err != nil || balance.Cmp(h.minimumBalance.ToInt()) < 0 {
+		balance, err := h.subscriptions.GetMaxUserBalance(sender)
+		if err != nil {
+			h.lggr.Debug("error while getting max user balance", "sender", msg.Body.Sender, "err", err)
+			return fmt.Errorf("error while getting max user balance: %w", err)
+		}
+		if balance == nil {
+			h.lggr.Debug("received a message from unknown user", "sender", msg.Body.Sender)
```

Comment:
Could you explain what is happening here? When does this occur? Does it happen if the sender has cancelled all their subscriptions?

Reply:
"sender has insufficient balance" was incorrect error, because the sender had sufficient balance (5 LINK).

Reply:
I reverted the change and added nil balance replacement

Reply:
For a node that lags/fails to refresh balances it's a correct error. I think that trying to distinguish between these cases will be more confusing to users and to us as well.

Reply:
Ok, so I reverted the changes.